### PR TITLE
send http headers in openUrl calls

### PR DESCRIPTION
**Pull request by devoncarew**
- send http headers in openUrl calls
- rename the `http_channels.dart` file to just `http.dart`

@pq

### CODE REVIEW

File: lib/src/channels/http.dart [hunk 1]
```diff
@@ -220,3 +223,11 @@ class _HttpOverrides extends HttpOverrides {
   HttpClient createHttpClient(SecurityContext context) =>
       LoggingHttpClient(context: context);
 }
+
+Map<String, String> _headersToMap(HttpHeaders headers) {
+  Map<String, String> map = {};
```
Comment (pq) [inline lib/src/channels/http.dart]: Maybe make this map `final`?

Comment (devoncarew) [reply lib/src/channels/http.dart]: done!

**Comment by devoncarew** [comment]
updated -

**Comment by devoncarew** [comment]
(I think you'll need to merge it)

**Pull request merged**